code/modules/hallucinations/hallucinations.dm

```diff
@@ -0,0 +1,55 @@
+#define NO_THOUGHT 1
+#define NO_EMOTE 2
+#define HEARING_DEPENDENT 3
+
+
+/datum/hallucination
+	var/mob/living/carbon/holder	//Who is hallucinating?
+	var/allow_duplicates = TRUE		//This is set to false for hallucinations with long durations or ones we do not want repeated for a time
+	var/duration = 10				//how long before we call end()
+	var/min_power = 0 				//mobs only get this hallucination at this threshold
+	var/max_power = INFINITY		//mobs don't get this hallucination if it's above this threshold. Used to weed out more common ones if you're super fucked up
```

Comment:
It might be worth making some basic defines for max_power and min_power. So something like this:
`HALLUCINATION_POWER_TINY 10` etc.